src/Microsoft.DotNet.Interactive.Tests/SubmissionSplittingTests.cs

```diff
@@ -76,5 +76,30 @@ public void RequestDiagnostics_can_be_split_into_separate_commands()
                 .Should()
                 .NotContain("#!time");
         }
+
+        [Fact]
+        public void Pound_r_can_immediately_follow_a_language_directive_and_nuget_restore_is_generated()
+        {
+            var code = @"
+#!csharp
+#r ""nuget: System.Text.Json""
+1+1
+";
+
+            using var kernel = new CompositeKernel
+            {
+                new CSharpKernel().UseNugetDirective(),
+                new FSharpKernel().UseNugetDirective(),
+            };
+
+            kernel.DefaultKernelName = "csharp";
+
+            var command = new SubmitCode(code);
+            var commands = kernel.SubmissionParser.SplitSubmission(command);
+
+            commands
+                .Should()
+                .ContainSingle<DirectiveCommand>(d => d.ParseResult?.CommandResult?.Command?.Name == "#!nuget-restore");
```

Comment:
This is really an implementation detail. I wonder if there's a cleaner way to test this.